web-components/src/components/form/BuyCreditsForm/BuyCreditsForm.tsx

```diff
@@ -0,0 +1,207 @@
+import { ChangeEvent, MouseEvent, Suspense, useState } from 'react';
+import { SubmitHandler } from 'react-hook-form';
+import { TextButton } from 'web-components/src/components/buttons/TextButton';
+import CheckboxLabel from 'web-components/src/components/inputs/new/CheckboxLabel/CheckboxLabel';
+import { CreditsAmount } from 'web-components/src/components/inputs/new/CreditsAmount/CreditsAmount';
+import { Radio } from 'web-components/src/components/inputs/new/Radio/Radio';
+import { RadioGroup } from 'web-components/src/components/inputs/new/RadioGroup/RadioGroup';
+import { Loading } from 'web-components/src/components/loading';
+import Title from 'web-components/src/components/typography/new/Title';
+import Form from 'web-marketplace/src/components/molecules/Form/Form';
+import { useZodForm } from 'web-marketplace/src/components/molecules/Form/hook/useZodForm';
+
+import { PaymentOptions } from './BuyCreditsForm.PaymentOptions';
+import { buyCreditsFormSchema, FormFields } from './BuyCreditsForm.schema';
+import { PAYMENT_OPTIONS, PaymentOptionsType } from './BuyCreditsForm.types';
+
+export function BuyCreditsForm({
+  creditVintages,
+  cryptoOptions,
+}: {
+  creditVintages: { date: string; credits: string }[];
+  cryptoOptions: { label: string; description?: string; linkTo: string }[];
+}) {
+  const [paymentOption, setPaymentOption] = useState<PaymentOptionsType>(
+    PAYMENT_OPTIONS.CARD,
+  );
+  const [advanceSettingsOpen, setAdvanceSettingsOpen] = useState(false);
+  const [cryptoPurchaseOption, setCryptoPurchaseOption] = useState<
+    string | undefined
+  >(cryptoOptions?.[0]?.label);
+  const [creditVintageOptions, setCreditVintageOptions] = useState<
+    string[] | []
+  >([]);
+
+  const toggleAdvancedSettings = (e: MouseEvent<HTMLElement>) => {
+    e.preventDefault();
+    setAdvanceSettingsOpen(prev => !prev);
+  };
+
+  const form = useZodForm({
+    schema: buyCreditsFormSchema,
+    defaultValues: {
+      amountCurrency: 1,
+      amountCredits: 1,
+      cryptoPurchaseOption,
+    },
+    mode: 'onBlur',
+  });
+
+  const handleOnSubmit: SubmitHandler<FormFields> = async data => {
+    await new Promise(resolve => setTimeout(resolve, 1000)); // do some async stuff
+    console.log('🚀 ~ data', data);
+  };
+
+  const handleCryptoPurchaseOptions = (e: ChangeEvent<HTMLInputElement>) => {
+    form.setValue('cryptoPurchaseOption', e.currentTarget.value);
+    setCryptoPurchaseOption(e.currentTarget.value);
```

Comment:
why do we need both `cryptoPurchaseOption` to be a form value and in state?

Reply:
good point